File: Data Structures and Algorithms/LeetCode/1502. Can Make Arithmetic Progression From Sequence.cs
public class Solution {
    public bool CanMakeArithmeticProgression(int[] arr) {
        
        Array.Sort(arr);
        int diff=0;
        for(int i=0;i<arr.Length-1;i++){
           if(i==0) diff=arr[i+1]-arr[i];
            if(arr[i+1]-arr[i]!=diff){
                
                 return false;
            }
            
                
        }
        return true;
    }
}

File: Data Structures and Algorithms/LeetCode/1502. Can Make Arithmetic Progression From Sequence.py
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        minnum = min(arr)
        maxnum = max(arr)
        difference = (maxnum-minnum)/(len(arr)-1)

        total = minnum
        for i in range(len(arr)-1):
            if total not in arr:
                return False
            total += difference
        return True
